src/functional/ffront/func_to_past.py

```diff
@@ -37,6 +37,8 @@ class ProgramParser(DialectParser[past.Program]):
     def _postprocess_dialect_ast(
         cls, output_node: past.Program, closure_vars: dict[str, Any], annotations: dict[str, Any]
     ) -> past.Program:
+        if "return" in annotations and not isinstance(None, annotations["return"]):
```

Comment:
This should be done in the PAST type deduction.

Reply:
it is not possible to do it in the type_deduction as the return type there is the deduced type. Additionally, checks to match the specified and deduced type for foast were done in func_to_foast